core/src/main/kotlin/gateway/handler/GuildEventHandler.kt

```diff
@@ -87,10 +87,11 @@ internal class GuildEventHandler(
 
     private suspend fun handle(event: GuildUpdate, shard: Int) {
         val data = GuildData.from(event.guild)
+        val old = cache.query<GuildData> { idEq(GuildData::id, event.guild.id) }.singleOrNull()
```

Comment:
Instead of a cache query eventually use the EntityProvider?
If not, thinking about my use case with distributed cache / own gateway impl, how would I populate this field?

Reply:
The goal is to supply what we know before the event happened. Im not sure I understand

Reply:
Thinking on my distributed setup, where Kord does not have any cache - How can I populate this field from an outside scope?

Reply:
Cache and EntitySupplier do not provide the same functionality. You're free to implement your own `DataEntryCache` and hook them up in the `KordBuilder`, though we are moving towards a new API that will make it easier to provide custom implementations and optimize queries.